src/v/tristate.h

```diff
@@ -122,3 +122,9 @@ class tristate {
     using underlying_t = std::variant<std::monostate, std::optional<T>>;
     underlying_t _value;
 };
+
+// Factory method so we don't have to code in the type
+template<typename T>
+tristate<T> make_tristate(T val) {
+    return tristate<T>(std::make_optional<T>(val));
+}
```

Comment:
We have CTAD, so a constructor would also work:
```c++
    constexpr explicit tristate(T t) noexcept
      : _value(std::move(t)) {}
```

Reply:
Ahh, it might create an ambiguity, don't bother spending time fixing that if its the case.